opm/simulators/wells/WellState.cpp

```diff
@@ -919,48 +919,44 @@ template<class Scalar>
 void WellState<Scalar>::communicateGroupRates(const Parallel::Communication& comm)
 {
     // Compute the size of the data.
-    std::size_t sz = 0;
-    for (const auto& [_, owner_rates] : this->well_rates) {
-        (void)_;
-        const auto& [__, rates] = owner_rates;
-        (void)__;
-        sz += rates.size();
-    }
+    std::size_t sz = std::accumulate(this->well_rates.begin(), this->well_rates.end(), std::size_t{0},
+                     [](const std::size_t acc, const auto& rates)
+                     { return acc + rates.second.second.size(); });
     sz += this->alq_state.pack_size();
 
     // Make a vector and collect all data into it.
     std::vector<Scalar> data(sz);
-    std::size_t pos = 0;
-    for (const auto& [_, owner_rates] : this->well_rates) {
-        (void)_;
-        const auto& [owner, rates] = owner_rates;
-        for (const auto& value : rates) {
-            if (owner)
-                data[pos++] = value;
-            else
-                data[pos++] = 0;
-        }
+    auto pos = data.begin();
+    std::for_each(this->well_rates.begin(), this->well_rates.end(),
+                  [&pos](const auto& input)
+                  {
+                      const auto& [owner, rates] = input.second;
+                      if (owner) {
+                          std::copy(rates.begin(), rates.end(), pos);
+                      }
+                      pos += input.second.second.size();
```

Comment:
You've already decomposed `input.second` into `owner` and `rates`, so maybe we could write this as
```C++
pos += rates.size();
```
instead?

Reply:
yup.